### PR TITLE
Bug 1853045 - Add nimbus homescreen survey message integration test - Dismiss survey

### PR DESCRIPTION
**Pull request by jrbenny35**
https://mozilla-hub.atlassian.net/browse/EXP-3818

### Pull Request checklist

- [✅] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [✅] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1853045

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt [hunk 1]
```diff
@@ -22,7 +22,12 @@ class SurveyExperimentIntegrationTest {
     private val experimentName = "Viewpoint"
 
     @get:Rule
-    val activityTestRule = HomeActivityTestRule()
+    val activityTestRule = HomeActivityTestRule(
+        isJumpBackInCFREnabled = false,
+        isPWAsPromptEnabled = false,
+        isTCPCFREnabled = false,
+        isDeleteSitePermissionsEnabled = true,
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt:29]: Not sure if you specifically wanted to enable this, as well as the disabling the others.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/test_integration.py [hunk 2]
```diff
@@ -5,3 +5,8 @@
 def test_survey_navigates_correctly(setup_experiment, gradlewbuild, load_branches):
     setup_experiment(load_branches)
     gradlewbuild.test("SurveyExperimentIntegrationTest#checkSurveyNavigatesCorrectly")
+
+@pytest.mark.parametrize("load_branches", [("branch")], indirect=True)
+def test_survey_no_thanks_navigates_correctly(setup_experiment, gradlewbuild, load_branches):
+    setup_experiment(load_branches)
+    gradlewbuild.test("SurveyExperimentIntegrationTest#checkSurveyNoThanksNavigatesCorrectly")
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/test_integration.py:12]: newline

File: fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt (hunk 1 above)
Comment (jrbenny35) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt:29]: which line exactly?

Comment (jrbenny35) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/experimentintegration/SurveyExperimentIntegrationTest.kt]: Okay, I removed some of those. I think the one's I kept are all I need.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 3]
```diff
@@ -1199,6 +1221,13 @@ class BrowserRobot {
             surveyButton.waitForExists(waitingTime)
             surveyButton.click()
 
+            BrowserRobot().interact()
+            return Transition()
+        }
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:1226]: Let's space out these functions here too

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -881,6 +892,17 @@ class BrowserRobot {
         button.waitForExists(waitingTime)
         button.click()
     }
+    fun clickNoThanksSurveyButton() {
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt:895]: nit: newline, let's space out these functions here and below too

**Review by AaronMT — APPROVED** [review]
Missed a minor newline between functions otherwise LGTM, please update before requesting landing

**Pull request merged**